### PR TITLE
Feat: work schema api

### LINKED ISSUE #42: Use Intracto API for work schedules
Remove holidays page as well

### PR DESCRIPTION
**Pull request by italoteix**
# Changes

## Related issues

Resolves #42 

## Added

- Work-scheme types
- Helper function to mark day as holiday based on work-scheme
- `isLeaveDay` added to `WeekDay` type
- Show indicators and Frey out leave days

## Removed

- Functions that automatically creates projects based on work-scheme results

## Changed

- No longer mocked work-scheme
- App also checks work-scheme to see if is holiday

## How to test

- Check if the holidays are correctly displayed
- Check if leave days are correctly displayed

## Screenshots

N/A

### CODE REVIEW

**Review by lassecapel — COMMENTED** [review]
What happens if a holiday is requested after an hour entry is already added? Is it removed from the timesheet then?

**Comment by italoteix** [comment]
> What happens if a holiday is requested after an hour entry is already added? Is it removed from the timesheet then?

Good question. It isn't.
Currently, we have non-blocking logic for holidays. 
So the employee still can add hours for holidays, although he'll get a warning before submit and the admin will review the timesheet.
So if the employee holiday request is before the review, the admin can either deny the timesheet or edit it himself.

File: services/work-scheme-service.ts [hunk 1]
```diff
@@ -1,61 +1,44 @@
...
-  getWorkScheme(params: {
-    employeeId: string;
+  axios: NuxtAxiosInstance;
+  ApiUrl = "/api/v1";
+
+  constructor(axios: NuxtAxiosInstance) {
+    this.axios = axios;
+  }
+
+  async getWorkScheme(params: {
```
Comment (mmehmetAliIzci) [inline services/work-scheme-service.ts:13]: Can we type the return type as `Promise<WorkScheme[] | undefined>`

File: store/records/actions.ts [hunk 2]
```diff
@@ -56,18 +60,23 @@ const actions: ActionTree<RecordsStoreState, RootStoreState> = {
...
+    let workSchemeResult: WorkScheme[] | undefined;
+    if (payload.bridgeUid) {
+      workSchemeResult = await this.app.$workSchemeService.getWorkScheme({
+        bridgeUid: payload.bridgeUid,
+        startDate: new Date(workWeek[0].date),
+        endDate: new Date(workWeek[6].date),
+      });
+    }
+
+    const holidays = rootState.holidays.holidays;
```
Comment (mmehmetAliIzci) [inline store/records/actions.ts:74]: inline ? 🤷

File: store/timesheets/getters.ts [hunk 3]
```diff
@@ -18,14 +18,13 @@ const getters: GetterTree<TimesheetsStoreState, RootStoreState> = {
     return employee ? employee.pendingTravelRecords : [];
   },
 
-  getEmployeePendingWeeks(state, _, rootState) {
+  getEmployeePendingWeeks(state) {
```
Comment (mmehmetAliIzci) [inline store/timesheets/getters.ts:21]: I cannot seem to find where does these getters has been used ? There seems to be no function call or not even name match on search

**Review by mmehmetAliIzci — CHANGES_REQUESTED** [review]

File: store/timesheets/getters.ts (hunk 3 above)
Comment (italoteix) [reply store/timesheets/getters.ts:21]: Probably a leftover. I'll delete this entire file

**Review by mmehmetAliIzci — APPROVED** [review]

**Pull request merged**